experimental/Rings/WeierstrassPreparationTheorem.jl

```diff
@@ -0,0 +1,240 @@
+##############################################################
+#
+#Conversion from Singular (weierstr.lib)
+#
+#############################################################
+
+export weierstrass_preparation_theorem
+
+export weierstrass_division_theorem
+
+
+"""
+`order_of_generality(f::MPolyElem, n::Int)`
+
+compute for a given polynomial `f` an integer `d` such that `f` is `x_n`-general of order `d` 
+with `x_n` the n-th variable of the polynomial ring of `f` 
+"""
+function order_of_generality(f::MPolyElem, n::Int)
+	 
+	vect = fill(parent(f)(0), nvars(parent(f))) 
+	vect[n] = gens(parent(f))[n] 
```

Comment:
```suggestion
	vect[n] = gen(parent(f), n)
```
`gen(R,i)` is more efficient than `gens(R)[i]`; so this should be also done in other places (e.g. a few lines below in this function)